web/pages/notifications.tsx

```diff
@@ -69,60 +167,326 @@ export default function Notifications() {
+  setNotificationsAsSeen(
+    notifications.filter((n) => hideNotificationIds.includes(n.id))
+  )
+  // Then remove them from the list of notifications to show
+  notifications = notifications.filter(
+    (notification) => !hideNotificationIds.includes(notification.id)
+  )
+
+  const notificationGroups: NotificationGroup[] = []
+  const notificationGroupsByDay = groupBy(notifications, (notification) =>
+    new Date(notification.createdTime).toDateString()
+  )
+  Object.keys(notificationGroupsByDay).forEach((day) => {
+    // Group notifications by contract:
+    const groupedNotificationsByContractId = groupBy(
+      notificationGroupsByDay[day],
+      (notification) => {
+        return notification.sourceContractId
+      }
+    )
+
+    // Create a notification group for each contract within each day
+    Object.keys(groupedNotificationsByContractId).forEach((contractId, i) => {
+      const notificationGroup: NotificationGroup = {
+        notifications: groupedNotificationsByContractId[contractId].sort(
+          (a, b) => {
+            return b.createdTime - a.createdTime
+          }
+        ),
+        sourceContractId: contractId,
+        isSeen: groupedNotificationsByContractId[contractId][0].isSeen,
+        timePeriod: day,
+      }
+      notificationGroups.push(notificationGroup)
+    })
+  })
+  return notificationGroups
+}
+
+function NotificationGroupItem(props: {
+  notificationGroup: NotificationGroup
+  className?: string
 }) {
-  const { notification, currentUser } = props
+  const { notificationGroup, className } = props
+  const { sourceContractId, notifications } = notificationGroup
+  const contract = useContract(sourceContractId ?? '')
+  const [activitySummaryLines, setActivitySummaryLines] = useState<string[]>([])
+  const numSummaryLines = 2
+
+  const summarizeMostRecentActivity = async (notifications: Notification[]) => {
+    let questionUpdates = 0
+    const tempSummaryLines: string[] = []
+    for (let i = 0; i < notifications.length; i++) {
+      if (tempSummaryLines.length + questionUpdates >= numSummaryLines) break
+      const notification = notifications[i]
+      if (!notification) continue
+      const { sourceType, sourceContractId, sourceId } = notification
+      if (!sourceId || !sourceContractId) continue
+      if (sourceType === 'comment' || sourceType === 'answer') {
+        const summary = await GetNotificationSummaryText(
+          sourceId,
+          sourceContractId,
+          sourceType
+        )
+        tempSummaryLines.push(
+          sourceType === 'answer' ? `New answer: ${summary}` : summary
+        )
+      } else if (sourceType === 'contract') {
+        questionUpdates += 1
+      }
+    }
+    if (questionUpdates > 0)
+      tempSummaryLines.push(
+        `${questionUpdates} new question update${
+          questionUpdates > 1 ? 's' : ''
+        }`
+      )
+    setActivitySummaryLines(tempSummaryLines)
+  }
+
```

Comment:
Suspicious that this isn't better solved by embedding the whole text of the notification + css to truncate it to one line

Reply:
I was just thinking about doing this esp after today's standup!

Reply:
hoooo wee I should've done this to begin with!